app/scripts/modules/core/src/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx

```diff
@@ -29,17 +33,66 @@ export class ManualJudgmentApproval extends React.Component<
       submitting: false,
       judgmentDecision: null,
       judgmentInput: {},
+      applicationRoles: {},
+      userRoles: [],
       error: false,
     };
   }
 
+  public componentDidMount() {
+      const applicationName = this.props.execution.application;
+      ApplicationReader.getApplicationPermissions(applicationName).then(result => {
+          if (result) {
+              this.setState({
+                  applicationRoles: result,
+              });
+          }
+      });
+      this.setState({
+          userRoles: AuthenticationService.getAuthenticatedUser().roles,
+      });
+  }
+
   private provideJudgment(judgmentDecision: string): void {
     const { application, execution, stage } = this.props;
     const judgmentInput: string = this.state.judgmentInput ? this.state.judgmentInput.value : null;
     this.setState({ submitting: true, error: false, judgmentDecision });
     ReactInjector.manualJudgmentService.provideJudgment(application, execution, stage, judgmentDecision, judgmentInput);
   }
 
+  private isManualJudgmentStageNotAuthorized(): boolean {
+      let isStageNotAuthorized = true;
+      const {
+          applicationRoles,
+          userRoles
+      } = this.state;
+      const stageRoles = this.props.stage.context.selectedStageRoles || [];
```

Comment:
Optional chaining since its nested to avoid an error being thrown
```suggestion
      const stageRoles = this.props.stage?.context?.selectedStageRoles || [];
```

Reply:
done..